src/Swarm/Web.hs

```diff
@@ -99,15 +111,22 @@ docsBS =
  where
   intro = SD.DocIntro "Swarm Web API" ["All of the valid endpoints are documented below."]
 
-mkApp :: IORef AppState -> Servant.Server SwarmAPI
-mkApp appStateRef =
+mkApp ::
+  -- | Readonly
+  IORef AppState ->
```

Comment:
It would be nice to encode this invariant in the types.  I wonder if there's a standard way to create a value which gives read-only access to an `IORef`.  Of course we could create our own opaque newtype which wraps `IORef`, and export only a function to read it but no constructor.  But maybe something like that already exists in a standard library somewhere?

Reply:
Couldn't find a standard library version, so I created `ReadableIORef`.